spring-integration-file/src/main/java/org/springframework/integration/file/remote/RemoteFileTemplate.java

```diff
@@ -434,6 +453,28 @@ public boolean get(final String remotePath, final InputStreamCallback callback)
 	}
 
 	@Override
+	public <T> T invoke(OperationsCallback<F, T> action) {
+		Session<F> contextSession = this.contextSessions.get();
+		if (contextSession == null) {
+			this.contextSessions.set(this.sessionFactory.getSession());
+		}
+		this.activeTemplateCallbacks.incrementAndGet();
```

Comment:
The `activeTemplateCallbacks` is simply an optimization to avoid accessing the thread local if no invokes are in process.